Tools/Carbonix_scripts/sitl_engine_curve_fit.py

```diff
@@ -0,0 +1,360 @@
+"""
+Engine Parameter Calibration Script for ArduPilot SITL
+
+This script is used to calibrate engine behavior parameters for ArduPilot SITL
+simulations. It handles calibration for cylinder head temperature (CHT and
+CHT2) and fuel consumption using flight log data. The calibration process
+optimizes key fit parameters to minimize the difference between actual and
+simulated values from the log data.
+
+Features:
+  - Independent calibration for CHT, CHT2, and fuel consumption.
+  - Visualization of actual vs simulated results for all three metrics.
+
+Input CSV File Format Requirements: The input file should be a CSV file with
```

Comment:
any particular reason you not directly processing a binary log? 

Also, putting comment on how to extract this info from binary would be handy.

Reply:
I generate the csv with UAVLogViewer. I plot all the variables of interest, and then hit the "export data" button.

Three big reasons:
1. Much easier to get resampled data this way, with all columns sampled at the same evenly-spaced time interval
2. Easier to select the region of interest that I want to fit.
3. Way less data to import when I was running repeatedly during dev.

Reply:
That make sense. Also didn't realised we can export from UAVlogViewer.